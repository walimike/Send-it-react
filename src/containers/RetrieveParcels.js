import React, { Component } from 'react';
import ParcelTable from '../components/parcels/Parceltable';
import axios from 'axios';

export default class RetrieveParcels extends Component {
  constructor(props) {
    super(props);
   this.state = {
      parcels:[]
    };
  }

  componentWillMount(){
    return axios
      .get("https://wali-sendit.herokuapp.com/v2/api/users/parcels", {
        headers :{accept: "application/json",
        "content-type": "application/json",
        Authorization: `Bearer ${sessionStorage.getItem("token")}`
      }})
      .then(re => this.setState({parcels:re.data.Parcel}))
  }

  render() {
    return (
      <div >
          <div className="card grey lighten-5 z-depth-1">
            <div className="card-content">
              <span className="card-title center">Parcels</span>
            </div>
            <div className="card-action">
            <ParcelTable parcels={this.state.parcels} />
            </div>
          </div>
          </div>
    )
  }
}

