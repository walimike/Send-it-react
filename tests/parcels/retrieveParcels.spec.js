
import React from 'react';
import { shallow, mount } from 'enzyme';
import moxios from 'moxios';

import RetrieveParcels from '../../src/containers/RetrieveParcels';

describe('Retrieve Parcel Container', () => {
  

  it('should render without crashing', () => {
    const wrapper = shallow(
        <RetrieveParcels />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render mount', () => {
    const wrapper = shallow(
        <RetrieveParcels />
    );
    wrapper.instance().setState({parcels:[{},{}]})
  });
})