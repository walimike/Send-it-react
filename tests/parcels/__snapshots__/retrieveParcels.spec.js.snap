// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Retrieve Parcel Container should render without crashing 1`] = `ShallowWrapper {}`;
